google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploy.java

```diff
@@ -42,141 +39,91 @@
+  public void deploy(@NotNull File stagingDirectory, @NotNull ProcessStartListener startListener) {
+    final StringBuilder rawDeployOutput = new StringBuilder();
 
-      copyFile(stagingDirectory, "app.yaml", appYamlPath);
-      copyFile(stagingDirectory, "Dockerfile", dockerFilePath);
-    } catch (IOException ex) {
-      logger.warn(ex);
-      callback.errorOccurred(GctBundle.message("appengine.deployment.error.during.staging"));
-      return null;
+    DefaultDeployConfiguration configuration = new DefaultDeployConfiguration();
+    configuration.setDeployables(
+        Collections.singletonList(new File(stagingDirectory, "app.yaml")));
+    configuration.setProject(deploymentConfiguration.getCloudProjectName());
+    configuration.setPromote(true);
+    if (!StringUtil.isEmpty(deploymentConfiguration.getVersion())) {
+      configuration.setVersion(deploymentConfiguration.getVersion());
     }
 
-    return stagingDirectory;
-  }
-
-  /**
-   * Perform a deployment from the given staging directory.
-   */
-  private void deploy(@NotNull File stagingDirectory) {
-    final StringBuilder rawDeployOutput = new StringBuilder();
-    CloudSdk sdk;
-
-    ProcessOutputLineListener stdErrListener = new ProcessOutputLineListener() {
+    ProcessOutputLineListener outputListener = new ProcessOutputLineListener() {
       @Override
-      public void outputLine(String output) {
-        consoleLogLn(output);
+      public void outputLine(String line) {
+        loggingHandler.print(line + "\n");
       }
     };
-
-    ProcessOutputLineListener stdOutListener = new ProcessOutputLineListener() {
+    ProcessOutputLineListener deployOutputListener = new ProcessOutputLineListener() {
       @Override
       public void outputLine(String output) {
         rawDeployOutput.append(output);
       }
     };
-
     ProcessExitListener deployExitListener = new DeployExitListener(rawDeployOutput);
 
-    try {
-      sdk = prepareExecution(stdErrListener, stdOutListener, deployExitListener);
-    } catch (AppEngineException ex) {
-      callback.errorOccurred(GctBundle.message("appengine.deployment.error"));
-      return;
-    }
-
-    DefaultDeployConfiguration configuration = new DefaultDeployConfiguration();
-    configuration.setDeployables(Collections.singletonList(new File(stagingDirectory, "app.yaml")));
-    configuration.setProject(deploymentConfiguration.getCloudProjectName());
-    configuration.setPromote(true);
-    if (!StringUtil.isEmpty(deploymentConfiguration.getVersion())) {
-      configuration.setVersion(deploymentConfiguration.getVersion());
-    }
-
+    CloudSdk sdk = helper.createSdk(
+        loggingHandler,
+        startListener,
+        outputListener,
+        deployOutputListener,
+        deployExitListener);
     CloudSdkAppEngineDeployment deployment = new CloudSdkAppEngineDeployment(sdk);
     deployment.deploy(configuration);
   }
 
+  AppEngineHelper getHelper() {
```

Comment:
Because they are only needed by the Runner classes. I didn't find it necessary to further expose them.